File: skills/astra-toolkit/clients/csharp/examples/tables-create-text-index-object-analyzer.cs
using DataStax.AstraDB.DataApi;
using DataStax.AstraDB.DataApi.Core;

namespace Examples;

public class Program
{
  static async Task Main()
  {
    // Get an existing table
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      System.Environment.GetEnvironmentVariable("ASTRA_DB_API_ENDPOINT"),
      System.Environment.GetEnvironmentVariable("ASTRA_DB_APPLICATION_TOKEN")
    );
    var table = database.GetTable("**TABLE_NAME**");

    // Index a column
    await table.CreateTextIndexAsync(
      "**INDEX_NAME**",
      "**TEXT_COLUMN_NAME**",
      Builders.TableIndex.Text(
        new AnalyzerOptions
        {
          Tokenizer = new TokenizerOptions
          {
            Name = "standard",
            Arguments = new Dictionary<string, object>(),
          },
          Filters = { "lowercase", "stop", "porterstem", "asciifolding" },
          CharacterFilters = new List<string>(),
        }
      )
    );
  }
}

File: skills/astra-toolkit/clients/python/examples/tables-create-text-index-object-analyzer.py
import os
from astrapy import DataAPIClient
from astrapy.info import TableTextIndexOptions

# Get an existing table
client = DataAPIClient()
database = client.get_database(
    os.environ["ASTRA_DB_API_ENDPOINT"], token=os.environ["ASTRA_DB_APPLICATION_TOKEN"]
)
table = database.get_table("**TABLE_NAME**")

# Index a column
table.create_text_index(
    "**INDEX_NAME**",
    column="**TEXT_COLUMN_NAME**",
    options=TableTextIndexOptions(
        analyzer={
            "tokenizer": {"name": "standard", "args": {}},
            "filters": [
                {"name": "lowercase"},
                {"name": "stop"},
                {"name": "porterstem"},
                {"name": "asciifolding"},
            ],
            "charFilters": [],
        },
    ),
)
